3.process_cfret_features/scripts/1.sc_quality_control.py

```diff
@@ -0,0 +1,241 @@
+plate_4_df["Z_Score_Area"] = zscore(plate_4_df["Nuclei_AreaShape_Area"])
+plate_4_df["Z_Score_Intensity"] = zscore(
+    plate_4_df["Nuclei_Intensity_IntegratedIntensity_Hoechst"]
+)
+
+# Set a threshold for considering outliers (number of standard deviations away from the mean)
+outlier_threshold = 2
+
+# Filter DataFrame for outliers
+nuclei_outliers_df = plate_4_df[
+    (plate_4_df["Z_Score_Area"].abs() > outlier_threshold)
+    & (plate_4_df["Z_Score_Intensity"].abs() > outlier_threshold)
+]
+
+# Print outliers to assess how it detected outliers
+print(nuclei_outliers_df.shape[0])
+# Print the range of outliers
+print("Outliers Range:")
+print("Area Min:", nuclei_outliers_df['Nuclei_AreaShape_Area'].min())
+print("Area Max:", nuclei_outliers_df['Nuclei_AreaShape_Area'].max())
+print("Intensity Min:", nuclei_outliers_df['Nuclei_Intensity_IntegratedIntensity_Hoechst'].min())
+print("Intensity Max:", nuclei_outliers_df['Nuclei_Intensity_IntegratedIntensity_Hoechst'].max())
+nuclei_outliers_df[
+    [
+        "Nuclei_AreaShape_Area",
+        "Nuclei_Intensity_IntegratedIntensity_Hoechst",
+        "Image_Metadata_Well",
+        "Image_Metadata_Site",
+        "Metadata_Nuclei_Location_Center_X",
+        "Metadata_Nuclei_Location_Center_Y",
+    ]
+].sort_values(by="Nuclei_AreaShape_Area", ascending=True).head()
+
+
+# ### Scatter plot of single-cells based on Nuclei Area and Intensity
+
+# In[4]:
+
+
+# Set the default value to 'inlier'
+plate_4_df['Outlier_Status'] = 'inlier'
+
+# Update the 'Outlier_Status' column based on the outliers DataFrame using index
+plate_4_df.loc[plate_4_df.index.isin(nuclei_outliers_df.index), 'Outlier_Status'] = 'outlier'
+
+# Create scatter plot
+plt.figure(figsize=(10, 6))
+sns.scatterplot(
+    data=plate_4_df,
+    x="Nuclei_AreaShape_Area",
+    y="Nuclei_Intensity_IntegratedIntensity_Hoechst",
+    hue="Outlier_Status",
+    alpha=0.6
+)
+
+# Add mean lines
+plt.axvline(
+    x=plate_4_df["Nuclei_AreaShape_Area"].mean(),
+    color="r",
+    linestyle="--",
+    label=f'Mean Area: {plate_4_df["Nuclei_AreaShape_Area"].mean():.2f}',
+)
+plt.axhline(
+    y=plate_4_df["Nuclei_Intensity_IntegratedIntensity_Hoechst"].mean(),
+    color="b",
+    linestyle="--",
+    label=f'Mean Intensity: {plate_4_df["Nuclei_Intensity_IntegratedIntensity_Hoechst"].mean():.2f}',
+)
+
+plt.title("Scatter Plot of Nuclei Area vs. Nuclei Integrated Intensity for Plate 4")
+plt.xlabel("Nuclei Area")
+plt.ylabel("Nuclei Integrated Intensity (Hoechst)")
+# Move legend to bottom right
+plt.legend(loc='upper right', bbox_to_anchor=(0.4, 1.0), prop={'size': 10})
+plt.show()
+
+
+# ## Identify mis-segmented cells due to high confluence
+
+# ### Perform z-scoring to identify cells outliers
+
+# In[5]:
+
+
+# Calculate Z-scores for the Cells Area
+z_scores = zscore(plate_4_df['Cells_AreaShape_Area'])
+
+# Set a threshold for Z-scores to find outliers below the mean (number of standard deviations away from the mean)
+threshold_z = -1
```

Comment:
what does a negative threshold mean?

Reply:
Negative threshold in this case is identifying outlier cases that are 1 standard deviation below the mean for this feature. I will include this in the code comment.